[openusd_internal] Fix macro expansion for PXR_PEGTL_NAMESPACE

This patch fixes an OpenUSD bug with namespace token-pasting. The existing code
failed to substitute the macro value 'PXR_INTERNAL_NS', rather accidentally
using the macro name itself for the namespace. We should upstream this patch.

--- pxr/base/pegtl/README.md
+++ pxr/base/pegtl/README.md
@@ -39,7 +39,9 @@ This file defines the namespace name that PEGTL uses:
 #include "pxr/pxr.h"
 
 #if PXR_USE_NAMESPACES
-#define PXR_PEGTL_NAMESPACE PXR_INTERNAL_NS ## _pegtl
+#define _impl_PASTE2(x, y) x ## y
+#define _impl_PASTE(x, y) _impl_PASTE2(x, y)
+#define PXR_PEGTL_NAMESPACE _impl_PASTE(PXR_INTERNAL_NS, _pegtl)
 #else
 #define PXR_PEGTL_NAMESPACE pxr_pegtl
 #endif
--- pxr/base/pegtl/pegtl/config.hpp
+++ pxr/base/pegtl/pegtl/config.hpp
@@ -9,7 +9,9 @@
 #include "pxr/pxr.h"
 
 #if PXR_USE_NAMESPACES
-#define PXR_PEGTL_NAMESPACE PXR_INTERNAL_NS ## _pegtl
+#define _impl_PASTE2(x, y) x ## y
+#define _impl_PASTE(x, y) _impl_PASTE2(x, y)
+#define PXR_PEGTL_NAMESPACE _impl_PASTE(PXR_INTERNAL_NS, _pegtl)
 #else
 #define PXR_PEGTL_NAMESPACE pxr_pegtl
 #endif
